Nautilus/Assets/ModPrefabCache.cs

```diff
@@ -89,27 +90,24 @@ private void Awake()
 
     public void EnterPrefabIntoCache(GameObject prefab)
     {
-        // Proper prefabs can never exist in the scene, so parenting them is dangerous and pointless. 
-        if (prefab.IsPrefab())
-        {
-            InternalLogger.Debug($"Game Object: {prefab} is a proper prefab. Skipping parenting for cache.");
-        }
-        else
-        {
-            prefab.transform.parent = _prefabRoot;
-            prefab.SetActive(true);
-        }
 
         var prefabIdentifier = prefab.GetComponent<PrefabIdentifier>();
 
-        if (prefabIdentifier == null)
+        if(prefabIdentifier == null)
         {
-            InternalLogger.Warn($"ModPrefabCache: prefab is missing a PrefabIdentifier component! Unable to add to cache.");
+            InternalLogger.Warn($"ModPrefabCache: prefab {prefab.name} is missing a PrefabIdentifier component! Unable to add to cache.");
             return;
         }
 
-        if (!Entries.ContainsKey(prefabIdentifier.classId))
+        // Proper prefabs can never exist in the scene, so parenting them is dangerous and pointless. 
+        if (prefab.IsPrefab())
+        {
+            InternalLogger.Debug($"Game Object: {prefab} is a proper prefab. Skipping parenting for cache.");
+        }
+        else if(!Entries.ContainsKey(prefabIdentifier.classId))
         {
+            prefab.transform.parent = _prefabRoot;
+            prefab.SetActive(true);
```

Comment:
While this appears to be much better than before and won't always just add an entry to the cache without much checks, I think clashing and duplicates is still possible, but it's now just "less" likely.

I suggest testing the example mod. that one has a duplicate on the copper clone iirc.